static/src/javascripts/projects/common/modules/identity/consents.js

```diff
@@ -137,14 +147,40 @@ const checkAllOptOuts = (): Promise<void> =>
 const uncheckAllOptIns = (): Promise<void> =>
     toggleInputsWithSelector(optInClassName, false);
 
+const showUnsubscribeConfirmation = (): Promise<void> => {
+    const fetchButton = (): Promise<HTMLButtonElement> =>
+        fastdom.read(() =>
+            document.querySelector(`.${unsubscribeButtonClassName}`)
+        );
+
+    const updateVisibilityAndShowMessage = (
+        elem: HTMLButtonElement
+    ): Promise<void> =>
+        fastdom.write(() => {
+            prependSuccessMessage(
+                UNSUBSCRIPTION_SUCCESS_MESSAGE,
+                ((elem.parentElement: any): HTMLElement)
```

Comment:
@SiAdcock this was required to make flow type check correctly and @walaura and I think it's bad... is there a better way?

Reply:
Flow gives you a type error because `Element.parentElement()` returns an `Element`, whereas `prependSuccessMessage` expects the second argument to be a `HTMLElement` (which is a subclass of `Element`). Flow can't infer that `elem.parentElement` would definitely be a `HTMLElement`, it can only infer that it would be an `Element`.

To get around this, you could update the expected type in the `prependSuccessMessage()` arguments form `HTMLElement` to `Element`.

Reply:
I changed all the prepend* args to require just the `Element` type, but it seems `parentElement` is nullable `?Element`, and I get the error: 

```
Error: static/src/javascripts/projects/common/modules/identity/consents.js:162
162:                 elem.parentElement
                     ^^^^^^^^^^^^^^^^^^ null or undefined. This type is incompatible with the expected param type of
 27:     location: Element
                   ^^^^^^^ Element. See: static/src/javascripts/projects/common/modules/identity/modules/prependMessage.js:27
```

Reply:
Right, so you’ll also have to do a null check on elem.parentElement before calling prependSuccessMessage

Reply:
Is it desirable to take in `Element` instead of `HTMLElement` in functions?

Reply:
As discussed, do whatever's easier 😄

Reply:
> do a null check on elem.parentElement before calling prependSuccessMessage

Done... It pretty cool/magical that flow is able to see the null check and allow this to pass.